modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySpiLogTest.java

```diff
@@ -0,0 +1,225 @@
+     * which can be accessed with 3's Discovery SPI.
+     * <p>
+     * For us to get multiple effective addresses for connection, we override
+     * {@link TcpDiscoverySpi#getEffectiveNodeAddresses(TcpDiscoveryNode)} method in
+     * {@link TestCustomEffectiveNodeAddressesSpi} to return custom collection of {@link InetSocketAddress}.
+     * We only need the custom collection once, when 3rd node is trying to probe the 2nd one. As the method is used in
+     * different other scenarios throughout the test, we restrain the method to deviate from expected behaviour only
+     * for the test case, that is when {@link TcpDiscoveryNode#discoveryPort()} is 47501.
+     * <p>
+     * By manipulating the collection order we can check the failing, successful and ignored connection logs at once.
+     *
+     * @throws Exception If failed.
+     * @see TcpDiscoverySpi#getEffectiveNodeAddresses(TcpDiscoveryNode)
+     * @see TestCustomEffectiveNodeAddressesSpi
+     */
+    @Test
+    public void testMultipleSocketConnectionLogMessage() throws Exception {
+        LogListener lsnr0 = LogListener.matches(s ->
+                s.contains("Connection check to node") && s.contains("result=success"))
+            .build();
+
+        LogListener lsnr1 = LogListener.matches(s ->
+                s.contains("Connection check to node") && s.contains("result=skipped"))
+            .build();
+
+        LogListener lsnr2 = LogListener.matches(s ->
+                s.contains("Connection check to node") && s.contains("result=failed"))
+            .build();
+
+        LogListener lsnr3 = LogListener.matches(s ->
+                s.contains("Connection check to previous node done"))
+            .build();
+
+        testLog.registerListener(lsnr0);
+        testLog.registerListener(lsnr1);
+        testLog.registerListener(lsnr2);
+        testLog.registerListener(lsnr3);
+
+        try {
+            startGrid(0);
+            startGrid(1);
+
+            TestCustomEffectiveNodeAddressesSpi spi = new TestCustomEffectiveNodeAddressesSpi();
+            nodeSpi.set(spi);
+
+            startGrid(2);
+
+            ((TcpDiscoverySpi)ignite(0).configuration().getDiscoverySpi()).brakeConnection();
+
+            assertTrue(waitForCondition(lsnr0::check, 10_000L));
+            assertTrue(waitForCondition(lsnr1::check, 10_000L));
+            assertTrue(waitForCondition(lsnr2::check, 10_000L));
+            assertTrue(waitForCondition(lsnr3::check, 10_000L));
+        }
+        finally {
+            stopAllGrids();
+        }
+    }
+
+    /**
+     * This test uses quiet closure of given {@link Socket} ignoring possible checked exception, which triggers
+     * the previous node to check the connection to the following surviving one.
+     *
+     * @throws Exception If failed.
+     * @see TcpDiscoverySpi#brakeConnection()
+     */
+    @Test
+    public void testCheckBrakeConnectionSuccessSocketConnectionLogMessage() throws Exception {
```

Comment:
I think this test is redundant. Isn't in includd in `testMultipleSocketConnectionLogMessage()`?